contributor/outline.md

```diff
@@ -1,64 +1,138 @@
+    - If you are contributing the fix to an issue, be sure to follow all guidance and instruction documented in the issue.
+    - You should understand the project well enough to know that your contribution makes sense overall and is not just your personal snowflake.
+  - Understands they are not the home owner and acts accordingly.
+    - Leave any sense of pride or entitlement at the door.
+    Everyone is working together and doing their best.
+    Be graceful, polite, and unassuming in all interaction with the host team.
+    - Feedback from the host team is not any kind of personal criticism to you.
+    Treat and value it as data to help your contribution become all the better.
+    - If the host team decides not to accept a contribution, don't take it personally.
+    Remember and respect that the host team has ultimate decision over the project.
+    They're the ones that are signing up to maintain it, after all!
+    Try to collaborate on an aspect of the contribution that you all agree on,
+    and look for another way to fill the rest of your needs.
+  
+## Mechanics of Contributing
+
+  - This section contains practical advice and gotchas to remember when making an InnerSource contribution.
+  - Preparing to work.
+    - Give yourself enough lead time.
+  The development, review, and deployment process for your contribution will likely take longer than similar processes on your own team.
+  Keep that in mind and plan to start early enough so that your work is available for you to leverage at the time you need it.
+    - If the host team is in a different time zone then keep that in mind and plan for it when interacting with them.
+    - For larger contributions, get agreement from the host team on the user/technical design of the contribution _before_ submitting a pull request.
+  This type of high-level, up-front agreeemment early will save time in rework or rejection of your pull request down the road.
+    - Understand how the host team would like you to reach out for help and questions while you're working.
+  - Development
+    - If you get stuck, try to unblock yourself by reading the project documentation or looking at other parts of the code in the project.
+    - After trying the above if you're still stuck then reach out to the host team for help.
+    It doesn't do either side any good for you to spin forever on a problem that could be resolved in a few-minute conversation with someone more knowledgable about the project.
+    It's OK to ask for help.
+    You aren't expected to be as knowledgable and proficient as you might be on the full-time projects that your team owns.
+    - Match the existing coding style, indentation, etc, even if it's not what you would normally do.
+    A guest contribution for a feature is not the time to introduce a new way of structuring or formatting project code.
+  - Submitting the pull request
+    - Test out your contribution before opening the pull request.
```

Comment:
Oh yes please do so! ;) 
Would we want to hint to the value of CI/automation somewhere? Goes with the "can a contributor attempt to mentor host teams" bullet point that used to be in the outline - might actually be an example to explain this bullet point very briefly.

Reply:
Added a line that mentions running these automated checks.

There are some bullets in **The Contributor Ethos** that give specific examples of things that a _Contributor_ can do to mentor host teams on the value of letting them contribute.

I've intentionally left out anything about the _Contributor_ teaching them how to run their project better outside of their individual contribution.  I feel like that's a more advanced behavior and that this _Contributor_ section of tehe learning path should just teach a beginner how to get in their patch.

Reply:
Valid argument. Maybe we should start a list of things that could go in a 102 version of the learning path.

Reply:
Open an issue in this project and just start putting comments on it?  We can decide what to do with them later.

Reply:
Opened #62 for this

Reply:
Thank you!